mutex.go

```diff
@@ -44,10 +44,12 @@ var (
 // Lock waits up to duration d for l.Lock() to succeed.
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
-	go func() {
+	ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
+	defer cancel()
+	go func(context.Context) {
 		l.Lock()
 		done <- struct{}{}
-	}()
+	}(ctx)
```

Comment:
but... I don't see how ctx is being used inside the goroutine. it seems like it isn't at all. am I missing something?